### PR TITLE
docs/developing.md: Update outdated steps

### PR DESCRIPTION
**Pull request by makubacki**
Update formatting to use ruff and provide some background info
about how the pip modules are installed.

### CODE REVIEW

File: docs/contributor/developing.md [hunk 1]
```diff
@@ -75,19 +75,35 @@
 
 1. Run a Basic Syntax/Lint Check (using flake8) and resolve any issues
 
-    ``` cmd
```
Comment (Javagedes) [inline docs/contributor/developing.md:78]: This is out of date. pydocstyle and flake8 were replaced with `ruff`. The new command that should be documented is `ruff edk2toollib` and `ruff` comes installed with the command `pip install -e .[dev]`

**Review by Javagedes — CHANGES_REQUESTED** [review]

File: docs/contributor/developing.md (hunk 1 above)
Comment (makubacki) [reply docs/contributor/developing.md:78]: Updated to mention the pip install command earlier in the file and replace `flake8` with `ruff`.

File: docs/contributor/developing.md [hunk 2]
```diff
@@ -73,21 +73,30 @@
...
+      * Note: Newer editors are very helpful in resolving source formatting errors. For example, in VSCode, you can
+              open the Python file and use `Alt+Shift+F` to auto format. See the [Ruff VSCode extension](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff)
+              for more information.
 
 2. Run a Basic Python docstring Check (using pydocstring) and resolve any issues
 
-    ``` cmd
-    pydocstyle edk2toollib
-    ```
+    * Run pydocstyle
```
Comment (Javagedes) [inline docs/contributor/developing.md:95]: `ruff` is used for the `pydocstyle` formatting now. Rule set "E" is for pydocstyle errors and rule set "W" is for pydocstyle errors.

Comment (makubacki) [reply docs/contributor/developing.md:95]: Updated with those details.

**Pull request merged**